ams/server/src/main/java/com/netease/arctic/server/DefaultOptimizingService.java

```diff
@@ -548,6 +548,7 @@ public void run() {
                           .forEach(task -> retryTask(task, queue)));
           if (isExpired) {
             LOG.info("Optimizer {} has been expired, unregister it", keepingTask.getOptimizer());
+            deleteResource(keepingTask.getOptimizer().getResourceId());
```

Comment:
The createResource() maybe called by user.
Is it proper that system auto deleting resources managed by user himself?

Reply:
the createResource is actually called by user,
and the deleteResource is also will be called by user, if user want the release the optimzier,

but if the optimizer instance failed by some unexpected casuse, the system will delete the instance auto, but will
not clean the resource record in the record, and this resource will never be used again and will cause dead dirty data in the db forever. 

so when the instance is expired and be deleted by system, the resource associate with this instance should also be cleand

Reply:
I don't quite agree with that.

The record left in db has its meaning, cause when the optimizer came back, AMS could identify this resource. 
What you said I think is a dashboard design issue, the optimizer page don't reveal inactive optimizers. I think we could improve in that way.
A good and simple princeple is who creates who deletes.

Reply:
thanks your comment jin, agree with it, and will check the lifetime of resource again ,and get one graceful idea the resolve it.